### PR TITLE
Empty shipment logger

### PR DESCRIPTION
**Pull request by uri**
This change was added in to find the source of an empty shipment being created.

### CODE REVIEW

**Comment by uri** [comment]
#### Output

This was taken from my development environment.

```
# Logfile created on 2016-02-22 15:15:43 -0500 by logger.rb/44203
I, [2016-02-22T15:15:43.140718 #62464]  INFO -- : {"order_id":536,"package_count":2,"order_state":"address","shipment_ids":[],"shipment_count":0,"action":"post-build_packages--packages","completed_at":null}
I, [2016-02-22T15:15:43.246906 #62464]  INFO -- : {"order_id":536,"package_count":2,"order_state":"address","shipment_ids":[],"shipment_count":0,"action":"post-optimize_distributor_split--sort_packages","completed_at":null}
I, [2016-02-22T15:15:43.249660 #62464]  INFO -- : {"order_id":536,"package_count":1,"order_state":"address","shipment_ids":[],"shipment_count":0,"action":"post-sort_packages--prioritized_packages","completed_at":null}
I, [2016-02-22T15:15:43.252898 #62464]  INFO -- : {"order_id":536,"package_count":1,"order_state":"address","shipment_ids":[],"shipment_count":0,"action":"post-adjust_packages--prioritized_packages","completed_at":null}
I, [2016-02-22T15:15:43.255817 #62464]  INFO -- : {"order_id":536,"package_count":1,"order_state":"address","shipment_ids":[],"shipment_count":0,"action":"post-prioritize_packages--packages","completed_at":null}
I, [2016-02-22T15:15:43.435740 #62464]  INFO -- : {"order_id":536,"package_count":1,"order_state":"address","shipment_ids":[],"shipment_count":0,"action":"post-estimate_packages--packages","completed_at":null}
I, [2016-02-22T15:15:50.277225 #62464]  INFO -- : {"order_id":536,"package_count":null,"order_state":"confirm","shipment_ids":[152],"shipment_count":1,"action":"commit_confirm!","completed_at":"2016-02-22T20:15:50Z"}

```

**Comment by uri** [comment]
@glongman do we want number of packages too?

File: core/app/models/spree/stock/coordinator.rb [hunk 1]
```diff
@@ -8,9 +8,14 @@ def initialize(order)
       end
 
       def packages
+        DebugLogger.log @order, action: "pre-build_packages--#{ __method__ }"
         packages = build_packages
+        DebugLogger.log @order, action: "post-build_packages--#{ __method__ }"
```
Comment (glongman) [inline core/app/models/spree/stock/coordinator.rb]: log package count here

File: core/app/models/spree/stock/coordinator.rb [hunk 2]
```diff
@@ -8,9 +8,14 @@ def initialize(order)
...
         packages = prioritize_packages(packages)
+        DebugLogger.log @order, action: "post-prioritize_packages--#{ __method__ }"
```
Comment (glongman) [inline core/app/models/spree/stock/coordinator.rb]: log package count here

File: core/app/models/spree/stock/coordinator.rb [hunk 3]
```diff
@@ -8,9 +8,14 @@ def initialize(order)
       end
 
       def packages
+        DebugLogger.log @order, action: "pre-build_packages--#{ __method__ }"
```
Comment (glongman) [inline core/app/models/spree/stock/coordinator.rb]: dont bother. just noise

File: core/app/models/spree/stock/coordinator.rb [hunk 4]
```diff
@@ -8,9 +8,14 @@ def initialize(order)
...
         packages = build_packages
+        DebugLogger.log @order, action: "post-build_packages--#{ __method__ }"
         packages = prioritize_packages(packages)
+        DebugLogger.log @order, action: "post-prioritize_packages--#{ __method__ }"
         packages = estimate_packages(packages)
+        DebugLogger.log @order, action: "post-estimate_packages--#{ __method__ }"
+        packages
```
Comment (glongman) [inline core/app/models/spree/stock/coordinator.rb:17]: log package count here

File: core/app/models/spree/stock/prioritizer.rb [hunk 5]
```diff
@@ -10,9 +10,13 @@ def initialize(order, packages, adjuster_class=Adjuster)
       end
 
       def prioritized_packages
+        DebugLogger.log @order, action: "pre-sort_packages--#{__method__}"
```
Comment (glongman) [inline core/app/models/spree/stock/prioritizer.rb]: noise

File: core/app/models/spree/stock/prioritizer.rb [hunk 6]
```diff
@@ -10,9 +10,13 @@ def initialize(order, packages, adjuster_class=Adjuster)
...
         sort_packages
+        DebugLogger.log @order, action: "post-sort_packages--#{__method__}"
```
Comment (glongman) [inline core/app/models/spree/stock/prioritizer.rb]: log package count here

File: core/app/models/spree/stock/prioritizer.rb [hunk 7]
```diff
@@ -10,9 +10,13 @@ def initialize(order, packages, adjuster_class=Adjuster)
...
         adjust_packages
+        DebugLogger.log @order, action: "post-adjust_packages--#{__method__}"
```
Comment (glongman) [inline core/app/models/spree/stock/prioritizer.rb]: log package count here

File: core/app/models/spree/stock/prioritizer.rb [hunk 8]
```diff
@@ -10,9 +10,13 @@ def initialize(order, packages, adjuster_class=Adjuster)
...
         prune_packages
+        DebugLogger.log @order, action: "post-prune_packages--#{__method__}"
```
Comment (glongman) [inline core/app/models/spree/stock/prioritizer.rb]: noisy. count is logged after the method returns - as per my comment

**Comment by glongman** [comment]
except for the last entry the shipment count is of no value.
package count is kind. 

Also don't care about the "before" log entries - just makes the log noisy

**Comment by uri** [comment]
@glongman I updated the example out in the original comment, see https://github.com/HealthWave/spree/pull/13#issuecomment-187344068 for most recent.

edit: hold on, something isn't right...
edit2: okay now the comment has the correct example output.

**Comment by glongman** [comment]
LGTM

**Pull request merged**